### PR TITLE
[Silabs] Move Wi-Fi interface files to the final structure

### PR DESCRIPTION
**Pull request by mkardous-silabs**
#### Descriptions
PR moves the Wi-Fi abstraction to the final directory structure and cleans up a few includes

#### Tests
CI & manual tests

### CODE REVIEW

**Review by lpbeliveau-silabs — APPROVED** [review]

File: src/platform/silabs/efr32/OTAImageProcessorImpl.cpp [hunk 1]
```diff
@@ -22,7 +22,7 @@
 #include <platform/silabs/SilabsConfig.h>
 
 #if SL_WIFI
-#include <platform/silabs/wifi/wf200/platform/spi_multiplex.h>
+#include <platform/silabs/wifi/ncp-common/spi_multiplex.h>
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply src/platform/silabs/efr32/OTAImageProcessorImpl.cpp]: done

File: src/platform/silabs/wifi/BUILD.gn [hunk 2]
```diff
@@ -40,7 +40,7 @@ if (chip_enable_wifi && !wifi_soc) {
   if (use_rs9116) {
     import("${silabs_platform_dir}/wifi/rs911x/rs911x.gni")
   } else if (use_SiWx917) {
-    import("${silabs_platform_dir}/wifi/rs911x/rs9117.gni")
+    import("${silabs_platform_dir}/wifi/SiWx/ncp-platform/rs9117.gni")
```
[the remark this answers is not in the document]
Comment (mkardous-silabs) [reply src/platform/silabs/wifi/BUILD.gn]: done

**Review by rcasallas-silabs — APPROVED** [review]

**Review by senthilku — APPROVED** [review]

**Review by ksperling-apple — APPROVED** [review]

File: src/platform/silabs/wifi/rs911x/ncp/rsi_wlan_config.h [hunk 3]
```diff
@@ -46,18 +46,9 @@
 
 //! To set Extended custom feature select bit map
 #if WIFI_ENABLE_SECURITY_WPA3_TRANSITION
-#ifdef CHIP_9117
```
Comment (rosahay-silabs) [inline src/platform/silabs/wifi/rs911x/ncp/rsi_wlan_config.h:49]: are we sure this is to be removed? I can see the XTAL_CLK and RAM_MODES to be different, are we maintaining that in a different location?

**Review by yunhanw-google — APPROVED** [review]

File: src/platform/silabs/wifi/rs911x/ncp/rsi_wlan_config.h (hunk 3 above)
Comment (mkardous-silabs) [reply src/platform/silabs/wifi/rs911x/ncp/rsi_wlan_config.h:49]: The file isn't included by any of the SiWx files at the Matter levels or by files in the Wifi SDK that we use so i think it is safe to remove.
From the blame, i think this was added in the early days of the 917 when it was still integrated to the 9116 APIs which it isn't anymore.

**Pull request merged**